### DISCUSSION TITLE
No Attribute and Unexpected Keyword argument error

### DISCUSSION THREAD

**Discussion by karthikraja95** [OpenMined/courses#277] [Course: Foundations of Private Computation]
I have been following the Federated Learning course and now I am in Concept 15 and 16, where we try to the max and min of the ages. The Openmined Notebook used `data_ptr.float().amax()` and `data_ptr.float().amin()`. It throws  **AttributeError: 'TensorPointer' object has no attribute 'amax'**

I resolved this by using `max()` and `min()` instead of `amax()` and `amin()`

I got **TypeError: add_handler() got an unexpected keyword argument 'name'** when adding request handler while executing the following code

`duet.requests.add_handler(name="result_download", action="accept")`

I am not sure why the Notebook code is not working on my local system. I definitely know that someone at OpenMinded tested these before releasing the code? Just wondering what the reason and how to solve this issue? 

Thanks
Karthik

**Comment by cbs-rschreijen** [comment]
I am having the same issue here. When I remove the `name="result_download"` parameter from all statements all code executes without error. I don't know if this will be the official fix but it looks like it as I noticed this pull requests which seems to address this issue similarly: https://github.com/OpenMined/courses/pull/278

**Comment by karthikraja95** [comment]
Yeah, I did that too, I removed it and it works. Maybe this might be an official fix, I don't know. Thanks for replying back though

**Comment by znreza** [comment]
Hi @karthikraja95 the codes were tested before launching for sure. However, some part of the API is getting changed which will be added in the next release soon. For the time being we are really sorry for some inconvenience you are having. Thanks for your patience.

[Here](https://github.com/OpenMined/courses/pull/278) is the PR for the 'name' parameter changed.
